assets/js/product.js

```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
             });
         }
 
+        function partyComparePropertyMaps(propMap1, master) {
+            return _.every(_.keys(master), function (x) {
+                var retVal = propMap1.hasOwnProperty(x);
+                if (retVal) {
+                    retVal = propMap1[x][0].value == master[x][0].value;
+                }
+                return retVal;
+            });
+        }
+
+
+        function filterVariationBySelectedProps(variations, selectedPropMap) {
```

Comment:
'filterVariationByProperties' is better than 'filterVariationBySelectedProps'

Reply:
maybe 'getVariationByPropertyMaps' is better than 'filterVariationByProperties' )

Reply:
renamed